src/components/download-overlay/download-overlay.tsx

```diff
@@ -28,98 +28,139 @@ interface DownloadOverlayProps extends AssetsListProps {
-        return (
-          <SourcesList
-            files={downloadMetadata!.flavors}
-            imageUrl={downloadMetadata!.imageDownloadUrl}
-            downloadPluginManager={downloadPluginManager}
-            downloadConfig={downloadConfig}
-            fileName={downloadMetadata!.fileName}
-            displayFlavors={downloadConfig.displayFlavors}
-          />
-        );
-      };
-
-      const renderCaptions = () => {
-        return (
-          <CaptionsList
-            files={downloadMetadata!.captions}
-            downloadPluginManager={downloadPluginManager}
-            fileName={downloadMetadata!.fileName}
-            shouldFocus={!shouldRenderSources}
-          />
-        );
-      };
-
-      const renderAttachments = () => {
-        return (
-          <AttachmentsList
-            files={downloadMetadata!.attachments}
-            downloadPluginManager={downloadPluginManager}
-            shouldFocus={!shouldRenderSources && !shouldRenderCaptions}
-          />
-        );
-      };
-
-      const shouldRenderSources = downloadConfig.displaySources && (downloadMetadata!.flavors.length || downloadMetadata!.imageDownloadUrl);
-      const shouldRenderCaptions = downloadConfig.displayCaptions && downloadMetadata!.captions.length;
-      const shouldRenderAttachments = downloadConfig.displayAttachments && downloadMetadata!.attachments.length;
-
-      return isVisible ? (
-        <OverlayPortal>
-          <FocusTrap active>
-            <Overlay
-              open
-              onClose={() => {
-                updateOverlay(false);
-                downloadPluginManager.setShowOverlay(false);
-              }}
-              type="playkit-download">
-              <div data-testid="download-overlay" className={styles.downloadOverlay}>
-                <div className={styles.header}>{downloadsLabel}</div>
-                <div className={styles.fileInfoList}>
-                  {shouldRenderSources || shouldRenderCaptions ? (
-                    <div className={styles.sourcesCaptionsContainer}>
-                      {shouldRenderSources && renderSources()}
-                      {shouldRenderCaptions && renderCaptions()}
-                    </div>
-                  ) : undefined}
-                  {shouldRenderAttachments && renderAttachments()}
+    withEventManager(
+      ({
+        downloadPluginManager,
+        eventManager,
+        downloadsLabel,
+        mainStream,
+        additionalStreams,
+        captionsLabel,
+        downloadMetadatas,
+        updateOverlay
+      }: DownloadOverlayProps) => {
+        const [isVisible, setIsVisible] = useState(false);
+        const mainSourceMetadata = downloadMetadatas[0];
+        // tslint:disable-next-line: ts-expect-error
+        // @ts-expect-error: Property 'config' is protected and only accessible within class 'BasePlugin' and its subclasses
+        const downloadConfig = downloadPluginManager.downloadPlugin.config;
```

Comment:
at this point, maybe just pass config inside a context if we want it to be accessible to components at more inner levels